### DISCUSSION TITLE
Shuffle random indexing Tx in-out for Solana (SOL)

### DISCUSSION THREAD

**Discussion by serrq** [horizontalsystems/unstoppable-wallet-android#5691] [Ideas]
It is a feature that I really appreciated in Bitcoin, that of having a different address for each new transaction.

I’d like to understand if the same thing can be done with the SOL wallet.

**Comment by abdrasulov** [comment]
Hi @serrq. 

This feature is specific to bitcoin-based blockchains. It is related to how the account balance is calculated. Bitcoin-based blockchain has unspent output transactions. The sum of these UTXOs is the balance of the account. And each UTXO may be owned by different addresses. As for Solana, the balance of an account is stored as its state.

**Reply by serrq** [reply]
Hi Abd. Can’t anything be done about it? I mean, any solution that prevents third parties from correlating on the person/balance?

**Reply by esengulov** [reply]
> Hi Abd. Can’t anything be done about it? I mean, any solution that prevents third parties from correlating on the person/balance?

I am afraid that's not possible for Solana due to the nature of the blockchain itself.

**Reply by serrq** [reply]
> I am afraid that's not possible for Solana due to the nature of the blockchain itself.

Ok, thanks for reply.
